repair/repair.hh

```diff
@@ -364,21 +364,24 @@ struct node_ops_cmd_request {
     std::unordered_map<gms::inet_address, std::list<dht::token>> bootstrap_nodes;
     // Optional field, list uuids of tables being repaired, set by repair cmd
     std::list<table_id> repair_tables;
+    std::unordered_set<locator::host_id> hosts_to_quarantine;
```

Comment:
> I know it's not recommended to perform topology changes during upgrade, but is it actually prohibited? What if, for example, one of our nodes fail during upgrade, and we want to remove it? Must we rollback first?
> 

I believe so.  @tgrabiec?

> If it's not prohibited - just not recommended - then it could potentially happen that some nodes don't have this field in `node_ops_cmd_request`, some do. What would happen then?

It's an optional field.  Since we start using it only when the feature is enabled we're safe on this front.

> 
> Also - we're gossiping this information with `QUARANTINED_HOSTS`. Isn't it redundant to put it here as well?

I don't trust gossip ;-)
Just kidding... it's not about passing the information to other nodes but rather telling them to start quarantining those hosts now.  If the node operation is aborted we don't want to put the existing nodes in the quarantine too early (especially for decommission.  for remove and replace, the nodes are already presumed dead so they can be quarantined early in principle, but maybe aborting those ops means that the might have been revived and the admin wants them back, maybe?)